add lab#1

ValeryShestakovv opened the pull request.

andrejkrasnov commented on MarvelApp/ViewController.swift:
```diff
@@ -57,30 +77,42 @@ class ViewController: UIViewController {
+                                  y: scrollView.frame.size.height / 2 + scrollView.contentOffset.y)
         if let indexPath = self.galleryCollectionView.indexPathForItem(at: centerPoint) {
-            let cell = (self.galleryCollectionView.cellForItem(at: indexPath)) as! GalleryCollectionViewCell
+            let cell = cells[indexPath.row]
```
тут все же не ячейка а item например

andrejkrasnov commented on MarvelApp/ViewController.swift:
```diff
@@ -57,30 +77,42 @@ class ViewController: UIViewController {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: GalleryCollectionViewCell.reuseId, for: indexPath) as! GalleryCollectionViewCell
+        cell.mainImageView.image = cells[indexPath.row].image
+        cell.textLable.text = cells[indexPath.row].name
+        cell.textLable.font = UIFont(name: "HelveticaNeue-Bold", size: 30)
```
шрифт лучше задавать при создании. Мало случаев когда шрифт динамический

andrejkrasnov commented on MarvelApp/ViewController.swift:
```diff
@@ -57,30 +77,42 @@ class ViewController: UIViewController {
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: GalleryCollectionViewCell.reuseId, for: indexPath) as! GalleryCollectionViewCell
```
fore unwrap это плохо надо безопастно разворачивать например через guard

andrejkrasnov commented on MarvelApp/ViewController.swift:
```diff
@@ -57,28 +51,29 @@ final class ViewController: UIViewController {
-        galleryCollectionView.topAnchor.constraint(equalTo: label.bottomAnchor, constant: 10).isActive = true
-        galleryCollectionView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
+
+        figure.snp.makeConstraints { make in
```
nit: там есть edges для таких случаев

andrejkrasnov commented on MarvelApp/ViewController.swift:
```diff
@@ -57,28 +51,29 @@ final class ViewController: UIViewController {
+ 
+        logo.snp.makeConstraints { make in
+            make.top.equalTo(view.snp.top).inset(20)
+            make.left.equalTo(view.snp.left).inset(90)
```
nit:
можно было make.left.right.equasToSupersiew().inset(90)

andrejkrasnov commented on MarvelApp/ViewController.swift:
```diff
@@ -57,28 +51,29 @@ final class ViewController: UIViewController {
+        }
+ 
+        logo.snp.makeConstraints { make in
+            make.top.equalTo(view.snp.top).inset(20)
```
а чего константы не вынес ?

andrejkrasnov commented on MarvelApp/ViewController.swift:
```diff
@@ -57,28 +51,29 @@ final class ViewController: UIViewController {
+
+        label.snp.makeConstraints { make in
+            make.top.equalTo(logo.snp.bottom).inset(20)
+            make.centerX.equalTo(view.snp.centerX)
```
если текст будет очень длинный выйдет за края экрана надо ограничения по бокам добавить

andrejkrasnov replied:
тогда и центрировать не надо будет

andrejkrasnov commented on Podfile:
```diff
@@ -5,6 +5,7 @@ target 'MarvelApp' do
   # Comment the next line if you don't want to use dynamic frameworks
   use_frameworks!
 
+  pod 'SnapKit'
```
версии в подах лучше обрезать той которую используешь 
pod 'SnapKit', '~> 5.6.0'

чтобы через время собрать можно было если файлик podfile.lock утерян

andrejkrasnov reviewed (approved):
мелочи надо поправить

ValeryShestakovv committed:
add enviroment

ValeryShestakovv committed:
added logo, text label, collectionView, background triangle

added marvel logo, text label, implemented a collection view with animated scrolling, created a triangle view to display the figure in the background.

ValeryShestakovv committed:
fix logo and collectionView insets

ValeryShestakovv committed:
fix podfile, add version pods
